config/reader.go

```diff
@@ -359,17 +368,17 @@ func FromReader(
 	r io.Reader,
 	logger logging.Logger,
 ) (*Config, error) {
-	return fromReader(ctx, originalPath, r, logger, true)
+	return fromReader(ctx, originalPath, r, logger, newRemoteReader)
 }
 
-// FromReader reads a config from the given reader and specifies
+// fromReader reads a config from the given reader and specifies
 // where, if applicable, the file the reader originated from.
 func fromReader(
 	ctx context.Context,
 	originalPath string,
 	r io.Reader,
 	logger logging.Logger,
-	shouldReadFromCloud bool,
+	rrFactory remoteReaderFactory,
 ) (*Config, error) {
```

Comment:
this function is the injection point. 

having a factory argument is ugly... basically we need to read some information from the config on disk to create a grpc connection to app (robot part id, secret, app address), and that reading happens here. 

curious if folks have ideas to improve this.

Reply:
i've been thinking about it a bit and not sure if there's anything easy we can do here and still allow us to test fromReader() end to end. We could possibly split fromReader up, so we can test each part separately and not have to instantiate the remote reader within the function.

but if we can't really make it less ugly, this is fine as well

Reply:
cool, thanks for taking the time to think about it